src/sentry/api/endpoints/project_rule_actions.py

```diff
@@ -110,7 +111,10 @@ def execute_future_on_test_event(
                     logger.warning(
                         "%s.test_alert.unexpected_exception", callback_name, exc_info=True
                     )
-                    break
+                    error_id = sentry_sdk.capture_exception(exc)
```

Comment:
I realize this will add a bit more noise to our sentry issues in the short term, so I may need to split this out into a separate exception type, or perhaps just an info level log? Either way, we probably want to add custom handling for these exceptions to reduce customer confusion when they see this response.

Reply:
TIL this function has a return value :O

Reply:
Yep! We use this in our base API exception handler and surface error IDs there as well, which is the only reason I know about this 😅 

https://github.com/getsentry/sentry/blob/4efffec1debc5228464ff6c08f9e3517145d02df/src/sentry/api/base.py#L334